Alignment/OfflineValidation/bin/Zmumumerge.cc

```diff
@@ -0,0 +1,310 @@
+  fitter->useChi2_ = useChi2;
+  fitter->initMean(xMean, xMin, xMax);
+  fitter->initSigma(sigma, sigmaMin, sigmaMax);
+  fitter->initSigma2(sigma2, sigma2Min, sigma2Max);
+  fitter->initAlpha(1.5, 0.05, 10.);
+  fitter->initN(1, 0.01, 100.);
+  fitter->initFGCB(0.4, 0., 1.);
+
+  fitter->initMean(91.1876, xMin, xMax);
+  fitter->initGamma(2.4952, 0., 10.);
+  fitter->gamma()->setConstant(kTRUE);
+  fitter->initMean2(0., -20., 20.);
+  fitter->mean2()->setConstant(kTRUE);
+  fitter->initSigma(1.2, 0., 5.);
+  fitter->initAlpha(1.5, 0.05, 10.);
+  fitter->initN(1, 0.01, 100.);
+  fitter->initExpCoeffA0(-1., -10., 10.);
+  fitter->initExpCoeffA1(0., -10., 10.);
+  fitter->initExpCoeffA2(0., -2., 2.);
+  fitter->initFsig(0.9, 0., 1.);
+  fitter->initA0(0., -10., 10.);
+  fitter->initA1(0., -10., 10.);
+  fitter->initA2(0., -10., 10.);
+  fitter->initA3(0., -10., 10.);
+  fitter->initA4(0., -10., 10.);
+  fitter->initA5(0., -10., 10.);
+  fitter->initA6(0., -10., 10.);
+  TCanvas* c1 = new TCanvas();
+  c1->Clear();
+  c1->SetLeftMargin(0.15);
+  c1->SetRightMargin(0.10);
+
+  fitter->fit(th1d_input, "breitWignerTimesCB", "exponential", xMin, xMax, false);
+
+  c1->Print(Form("%s/fitResultPlot/%s_oldtool.pdf", output_path.Data(), s_name.Data()));
+  c1->Print(Form("%s/fitResultPlot/%s_oldtool.root", output_path.Data(), s_name.Data()));
+
+  FitOut fitRes(
+      fitter->mean()->getValV(), fitter->mean()->getError(), fitter->sigma()->getValV(), fitter->sigma()->getError());
+  return fitRes;
+}
+void Draw_th1d(TH1D* th1d_input, TString variable_name) {
+  TCanvas* c = new TCanvas();
+  c->cd();
+  gStyle->SetOptStat(0);
+  th1d_input->SetMarkerStyle(kFullCircle);
+  th1d_input->SetMarkerColor(kRed);
+  th1d_input->SetLineColor(kRed);
+  th1d_input->SetMaximum(91.4);
+  th1d_input->SetMinimum(90.85);
+  th1d_input->GetXaxis()->SetTitle(variable_name.Data());
+  th1d_input->GetXaxis()->SetTitleOffset(1.2);
+  th1d_input->GetYaxis()->SetTitle("Mass mean (GeV)");
+  th1d_input->Draw();
+  tlxg->DrawLatexNDC(0.2, 0.8, Form("%s", GT.Data()));
+  c->Print(Form("%s/fitResultPlot/mass_VS_%s.pdf", GT.Data(), variable_name.Data()));
+}
+
+const static int variables_number = 8;
+const TString tstring_variables_name[variables_number] = {
+    "CosThetaCS", "DeltaEta", "EtaMinus", "EtaPlus", "PhiCS", "PhiMinus", "PhiPlus", "Pt"};
+void Fitting_GetMassmeanVSvariables(TString inputfile_name, TString output_path) {
+  TH2D* th2d_mass_variables[variables_number];
+  TFile* inputfile = TFile::Open(inputfile_name.Data());
+  TDirectoryFile* tdirectory = (TDirectoryFile*)inputfile->Get("myanalysis");
+  for (int i = 0; i < variables_number; i++) {
+    TString th2d_name = Form("th2d_mass_%s", tstring_variables_name[i].Data());
+    th2d_mass_variables[i] = (TH2D*)tdirectory->Get(th2d_name);
+  }
+
+  gSystem->Exec(Form("mkdir -p %s", output_path.Data()));
+  gSystem->Exec(Form("mkdir -p %s/fitResultPlot", output_path.Data()));

```

Comment:
It would be better to use the `filesystem` library to create directories and handle files in general. You are actually loading the header of `boost::filesystem`. Here it would be sth like:
```c++
using fs = boost::filesystem; // after the headers
// ...
fs::path p = output_path.Data();
p /= "fitResultPlot";
fs::create_directories(p);
assert(fs::exists(p));
```